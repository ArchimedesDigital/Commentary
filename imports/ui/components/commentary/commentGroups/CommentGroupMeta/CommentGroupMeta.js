import React from 'react';
import AvatarIcon from '/imports/ui/components/avatar/AvatarIcon'; // eslint-disable-line import/no-absolute-path

const getWorkTitle = (commentGroup) => {
	let workTitle = commentGroup.work.title;
	if (workTitle === 'Homeric Hymns') {
		workTitle = 'Hymns';
	}
	return workTitle;
};

const CommentGroupMeta = ({ hideLemma, commentGroup }) => (
	<div className="comment-group-meta">
		{hideLemma === false ?
			<div className="comment-group-meta-inner comment-group-meta-ref">
				<div className="comment-group-ref">
					<span className="comment-group-ref-above">
						{getWorkTitle(commentGroup)} {commentGroup.subwork.title}
					</span>
					<h2 className="comment-group-ref-below">
						{commentGroup.lineFrom}{commentGroup.lineTo ? `-${commentGroup.lineTo}` : '' }
					</h2>

				</div>
				<div className="comment-group-commenters">

					{commentGroup.commenters.map(commenter => (
						<div
							key={commenter._id}
							className="comment-author"
							data-commenter-id={commenter._id}
						>
							<span className="comment-author-name">
								{commenter.name}
							</span>
							<a
								className="comment-author-image-wrap paper-shadow"
								href={`/commenters/${commenter.slug}`}
							>
								<AvatarIcon
									avatar={
										(commenter && 'avatar' in commenter) ?
										commenter.avatar.src
										: null
									}
								/>
							</a>
						</div>
					))}
				</div>
			</div>
		: '' }
	</div>
);
CommentGroupMeta.propTypes = {
	hideLemma: React.PropTypes.bool,
	commentGroup: React.PropTypes.shape({
		subwork: React.PropTypes.shape({
			title: React.PropTypes.string.isRequired,
		}),
		lineFrom: React.PropTypes.number.isRequired,
		lineTo: React.PropTypes.number,
		commenters: React.PropTypes.arrayOf(React.PropTypes.shape({
			_id: React.PropTypes.string.isRequired,
			name: React.PropTypes.string.isRequired,
			slug: React.PropTypes.string.isRequired,
			avatar: React.PropTypes.shape({
				src: React.PropTypes.string,
			})
		}))
	}),
};
CommentGroupMeta.defaultProps = {
	hideLemma: false,
	commentGroup: null,
};

export default CommentGroupMeta;
