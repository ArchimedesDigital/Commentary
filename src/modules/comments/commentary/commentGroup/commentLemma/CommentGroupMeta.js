import React from 'react';
import PropTypes from 'prop-types';
import AvatarIcon from '../../../../profile/avatar/AvatarIcon';

const getWorkTitle = (commentGroup) => {
	let workTitle = commentGroup.work.title;
	if (workTitle === 'Homeric Hymns') {
		workTitle = 'Hymns';
	}
	return workTitle;
};

const getLineTo = (commentGroup) => {
	if (commentGroup.lineTo === commentGroup.lineFrom || !commentGroup.lineTo) {
		return '';
	}
	return `- ${commentGroup.lineTo}`;
};

const CommentGroupMeta = ({ hideLemma, commentGroup }) => (
	<div className="comment-group-meta">
		{hideLemma === false ?
			<div className="comment-group-meta-inner comment-group-meta-ref">
				<div className="comment-group-ref">
					<span className="comment-group-ref-above">
						{getWorkTitle(commentGroup)} {commentGroup.subwork.title}
					</span>
					<h2 className="comment-group-ref-below">
						{commentGroup.lineFrom}{getLineTo(commentGroup)}
					</h2>

				</div>
				<div className="comment-group-commenters">

					{Object.keys(commentGroup.commenters).map(key => (
						<div
							key={commentGroup.commenters[key]._id}
							className="comment-author"
							data-commenter-id={commentGroup.commenters[key]._id}
						>
							<span className="comment-author-name">
								{commentGroup.commenters[key].name}
							</span>
							<a
								className="comment-author-image-wrap paper-shadow"
								href={`/commenters/${commentGroup.commenters[key].slug}`}
							>
								<AvatarIcon
									avatar={
										(commentGroup.commenters[key] && commentGroup.commenters[key].avatar) ?
										commentGroup.commenters[key].avatar.src
										: null
									}
								/>
							</a>
						</div>
					))}
				</div>
			</div>
		: '' }
	</div>
);
CommentGroupMeta.propTypes = {
	hideLemma: PropTypes.bool,
	commentGroup: PropTypes.shape({
		subwork: PropTypes.shape({
			title: PropTypes.string.isRequired,
		}),
		lineFrom: PropTypes.number.isRequired,
		lineTo: PropTypes.number,
		commenters: PropTypes.objectOf(PropTypes.shape({
			_id: PropTypes.string.isRequired,
			name: PropTypes.string.isRequired,
			slug: PropTypes.string.isRequired,
			avatar: PropTypes.shape({
				src: PropTypes.string,
			})
		}))
	}),
};
CommentGroupMeta.defaultProps = {
	hideLemma: false,
	commentGroup: null,
};

export default CommentGroupMeta;
