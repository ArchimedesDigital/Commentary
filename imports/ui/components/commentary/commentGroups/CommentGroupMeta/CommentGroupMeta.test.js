import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommentGroupMeta from './CommentGroupMeta';

describe('CommentGroupMeta', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<CommentGroupMeta
				commentGroup={{
					subwork: {
						title: 'testTitle'
					},
					lineFrom: 1,
					lineTo: 2,
					commenters: [
						{
							_id: 'testId',
							name: 'testName',
							slug: 'testName',
							avatar: {
								src: 'testSrc'
							}
						}
					]
				}}
			/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
