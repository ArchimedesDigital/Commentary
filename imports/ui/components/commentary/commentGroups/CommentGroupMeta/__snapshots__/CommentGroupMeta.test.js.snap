// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentGroupMeta renders correctly 1`] = `
<div
  className="comment-group-meta"
>
  <div
    className="comment-group-meta-inner comment-group-meta-ref"
  >
    <div
      className="comment-group-ref"
    >
      <span
        className="comment-group-ref-above"
      >
        testTitle
         
        testTitle
      </span>
      <h2
        className="comment-group-ref-below"
      >
        1
        - 2
      </h2>
    </div>
    <div
      className="comment-group-commenters"
    >
      <div
        className="comment-author"
        data-commenter-id="testId"
      >
        <span
          className="comment-author-name"
        >
          testName
        </span>
        <a
          className="comment-author-image-wrap paper-shadow"
          href="/commenters/testName"
        >
          <div
            alt="icon"
            className="avatar-icon"
            style={
              Object {
                "backgroundImage": "url(testSrc)",
              }
            }
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
